python/tutorials/06-fused-attention.py

```diff
@@ -144,7 +147,7 @@ def _attn_fwd(
     qk_scale = sm_scale * 1.44269504
     # load q: it will stay in SRAM throughout on NV GPUs but in VGPRs on AMD GPUs
     q = tl.load(Q_block_ptr)
-    q = (q * qk_scale).to(tl.float16)
+    q = (q * qk_scale).to(q.dtype)
```

Comment:
This is converting fp32 to fp8 when q is fp8. Is this handled by hw instruction? Since I did not see sw conversions between f32 and fp8.

Reply:
You can check the code here: https://github.com/ROCmSoftwarePlatform/triton/blob/1a8d529483171fbf03fdabf1d4dab3e6585a5a90/lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp#L1428-L1443

Specifically, we use  two steps for conversion from fp32 --> fp8 as:  fp32-->fp16, fp16-->fp8. The above code lines are for that. I am working on using hardware instructions to do type conversion for out fp8 types now, I think we should use the HW instruction for the type conversion here (much more efficient).